dev/planning-example.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# ============================
+# Observation Planning Toolbox
+# ============================
+"""
+The observation planning toolbox consists of a number of entities that
+simplify the planning and scheduling of observations by abstracting much
+of the boilerplate code necessary to accomplish the same thing using
+core astropy or ephemerides objects.  These can be thought of as common
+idioms or convenience classes and functions that would be recreated over
+and over by astronomers and or software engineers building telescope
+scheduling software.
+
+Important assumptions:
+
+- It should be convenient to specify dates and times in the local timezone
+  of the observer.
+
+- It should be convenient to do timezone conversions easily.
+
+  This means methods that can take dates can either take astropy
+  dates (as used internally) or python datetime objects (not naive, but
+  tagged with timezone).  So dates can be passed in as timezone-aware
+  datetime objects and results can be easily converted between such.
+
+"""
+
+# ================
+# Observer objects
+# ================
+"""
+An observer defines a single terrestrial location and applicable
+environmental attributes thereof that affect observation calculations
+at that location.
+"""
+# Define an observer by full specification
+#
+# Because there may be many attributes for an observer, I suggest using
+# keyword parameters for most of them.
+from astroplan.entity import Observer
+
+obs = Observer(name='Subaru Telescope',
+               longitude='-155:28:48.900',
```

Comment:
I pretty much agree with @cdeil here.  Basically, I think of the initializer for these classes as providing a convenient API for creating groups of other objects.

One thing I think we want to maintain rigorously, though: it should be as straightforward as possible how the keywords map to the internally stored objects.  So, e.g. for `Observer`, on creation the object should gain two attributes which are part of the public API: `obs.earthlocation` and `obs.altaz`.  With that in mind, a third creation mechanism should be possible, like this: ``obs = Observer(EarthLocation(...), AltAz(...))``, which of course just skips the parsing of the relevant keywords and just sets `earthlocation` and `altaz` directly.